app/assets/javascripts/thyme/utility.js

```diff
@@ -0,0 +1,146 @@
+     representation "#xxyyzz") such that e.g. black becomes dark grey.
+     The higher the value of "factor" the brighter the colors become.
+   */
+  lightenUp: function(color, factor) {
+    const red = Math.floor(((factor - 1) * 255 + Number("0x" + color.substr(5, 2))) / factor);
+    const green = Math.floor(((factor - 1) * 255 + Number("0x" + color.substr(3, 2))) / factor);
+    const blue = Math.floor(((factor - 1) * 255 + Number("0x" + color.substr(1, 2))) / factor);
+    return "#" + thymeUtility.toHexaDecimal(blue) +
+                 thymeUtility.toHexaDecimal(green) +
+                 thymeUtility.toHexaDecimal(red);
+  },
+
+  /*
+    Installs a listener which lets the video play/pause when clicked.
+   */
+  playOnClick() {
+    const video = thymeAttributes.video;
+    video.addEventListener('click', function() {
+      if (video.paused === true) {
+        video.play();
+      } else {
+        video.pause();
+      }
+    });
+  },
+
+  /*
+    Renders latex in a given HTML element.
+   */
+  renderLatex: function(element) {
+    renderMathInElement(element, {
+      delimiters: [
+        {
+          left: '$$',
+          right: '$$',
+          display: true
+        }, {
+          left: '$',
+          right: '$',
+          display: false
+        }, {
+          left: '\\(',
+          right: '\\)',
+          display: false
+        }, {
+          left: '\\[',
+          right: '\\]',
+          display: true
+        }
+      ],
+      throwOnError: false
+    });
+  },
+
+  /*
+    Convert time in seconds to string of the form H:MM:SS.
+   */
+  secondsToTime: function(seconds) {
+    let date = new Date(null);
+    date.setSeconds(seconds);
+    return date.toISOString().substr(12, 7);
+  },
+
+  /*
+    Sets up the label on the right side of the seek bar which displays
+    the maximum time of the video.
+    (In order to make this work, we have to wait for the video's metadata
+    to be loaded.)
+   */
+  setUpMaxTime(maxTimeId) {
+    const video = thymeAttributes.video;
+    video.addEventListener('loadedmetadata', function() {
+      const maxTime = document.getElementById(maxTimeId);
+      maxTime.innerHTML = thymeUtility.secondsToTime(video.duration);
+      if (video.dataset.time != null) {
+        const time = video.dataset.time;
+        video.currentTime = time;
+      }
+    });
+  },
+
+  /*
+    Converts a json timestamp into a double value containing the absolute value of seconds.
+   */
+  timestampToSeconds: function(timestamp) {
+    return 3600 * timestamp.hours + 60 * timestamp.minutes + timestamp.seconds + 0.001 * timestamp.milliseconds;
+  },
+
+  /*
+    Converts a given integer between 0 and 255 into a hexadecimal, s.t. e.g. "15" becomes "0f"
+    (instead of just "f") -> needed for correct format.
+   */
+  toHexaDecimal: function(int) {
+    if (int > 15) {
+      return int.toString(16);
+    } else {
+      return "0" + int.toString(16);
+    }
+  },
```

Comment:
That's nice!

As all functions in the utility object are written down as `doSth: function() { }`, I'd prefer to write it down like this for consitency:
`toHexaDecimal: function(int) {`
 ` 	return int.toString(16).padStart(2, '0');`
 ` },`
But I'm open minded for your solution :).

Reply:
Sure, the way you adapted it is fine. But I stumbled on the `int` as variable name as in many languages this is actually a reserved keyword (not in JS, but still, it might lead to confusion). So why not call it `integer`, `intValue` or something along these lines. Of course, this one is very nitpicky :)